addons/point_of_sale/models/pos_order.py

```diff
@@ -190,10 +191,10 @@ def _prepare_tax_base_line_values(self):
     @api.model
     def _get_invoice_lines_values(self, line_values, pos_line, move_type):
         # correct quantity sign based on move type and if line is refund.
-        is_refund = line_values.get('is_refund', False)
+        is_refund_order = pos_line.order_id.amount_total < 0.0
```

Comment:
Note to the reviewer: The negative discount line was considered as a refund. With the `qty_sign` below, it was changing the sign of the negative line to be positive and then, the discount was added to the invoice instead of subtracted. To me, it's a bug that needs to be fixed in saas-18.2 @caburj

Reply:
Indeed. I'm making the fix here: https://github.com/odoo/odoo/pull/201385

Reply:
why not use the refunded_order_id fields instead?